clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -61,24 +58,22 @@ public HubConnection(String url, Transport transport, Logger logger, boolean ski
                 }
             }
 
-            HubMessage[] messages = protocol.parseMessages(payload);
+            HubMessage[] messages = protocol.parseMessages(payload, connectionState);
 
             for (HubMessage message : messages) {
-                logger.log(LogLevel.Debug, "Received message of type %s", message.getMessageType());
+                logger.log(LogLevel.Debug, "Received message of type %s.", message.getMessageType());
                 switch (message.getMessageType()) {
                     case INVOCATION:
                         InvocationMessage invocationMessage = (InvocationMessage) message;
                         if (handlers.containsKey(invocationMessage.target)) {
```

Comment:
We don't need this check right? [`.get` returns `null` if the key doesn't exist](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#get-java.lang.Object-) so we're just checking twice.